dace/sdfg/replace.py

```diff
@@ -0,0 +1,92 @@
+""" Contains functionality to perform find-and-replace of symbols in SDFGs. """
+
+from dace import dtypes, properties, symbolic
+from dace.frontend.python.astutils import ASTFindReplace
+import re
+import sympy as sp
+from typing import Any, Union
+import warnings
+
+
+def _replsym(symlist, symrepl):
+    """ Helper function to replace symbols in various symbolic expressions. """
+    if symlist is None:
+        return None
+    if isinstance(symlist, (symbolic.SymExpr, symbolic.symbol, sp.Basic)):
+        return symlist.subs(symrepl)
+    for i, dim in enumerate(symlist):
+        try:
+            symlist[i] = tuple(
+                d.subs(symrepl) if symbolic.issymbolic(d) else d for d in dim)
+        except TypeError:
+            symlist[i] = (dim.subs(symrepl)
+                          if symbolic.issymbolic(dim) else dim)
+    return symlist
+
+
+def replace(subgraph: 'dace.sdfg.state.StateGraphView', name: str,
+            new_name: str):
+    """ Finds and replaces all occurrences of a symbol or array in the given
+        subgraph.
+        :param subgraph: The given graph or subgraph to replace in.
+        :param name: Name to find.
+        :param new_name: Name to replace.
+    """
+    symrepl = {
+        symbolic.symbol(name):
+        symbolic.symbol(new_name) if isinstance(new_name, str) else new_name
+    }
+
+    # Replace in node properties
+    for node in subgraph.nodes():
```

Comment:
How does this work for nested SDFGs?

Reply:
No need to replace into nested SDFGs, as the `symbol_mapping` ensures separate symbol systems and the data containers are different.